File: OSX/MacAudioStream/ViewController.m
//
//  ViewController.m
//  MacAudioStream
//
//  Created by MKolesov on 22/12/15.
//  Copyright © 2015 Michael Kolesov. All rights reserved.
//

#import "ViewController.h"
#import "AudioStreamEngine.h"

@interface ViewController () <AudioStreamEngineDelegate>
@property (weak) IBOutlet NSTextField *titleLabel;
@property (nonatomic, strong) AudioStreamEngine *audioStreamEngine;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.audioStreamEngine = [AudioStreamEngine sharedInstance];
    self.audioStreamEngine.delegate = self;
    
    // Do any additional setup after loading the view.
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

- (IBAction)startAction:(id)sender {
    
    if (self.audioStreamEngine.playing == NO) {
        [self.audioStreamEngine startWithURL:@"http://air.radiorecord.ru:8101/rr_128"];
    }
    
}

- (IBAction)stopAction:(id)sender {
    [self.audioStreamEngine stop];
}

#pragma mark - AudioStreamEngineDelegate

-(void) audioStreamEngineDidUpdateTitle:(NSString *)title
{
    [self.titleLabel setStringValue:title];
}

@end

File: OSX/MacAudioStream/ViewController.swift
//
//  ViewController.swift
//  MacAudioStream
//
//  Created by MKolesov on 25/12/15.
//  Copyright © 2015 Michael Kolesov. All rights reserved.
//

import Cocoa

class ViewController: NSViewController, AudioStreamEngineDelegate {

    @IBOutlet weak var titleLabel: NSTextField!
    
    var audioStream: AudioStreamEngine!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        audioStream = AudioStreamEngine.sharedInstance()
        audioStream.delegate = self
    }

    override var representedObject: AnyObject? {
        didSet {
        // Update the view, if already loaded.
        }
    }

    @IBAction func playAction(sender: AnyObject) {
        
        print("play")
        if(audioStream.playing) {
            return
        }
        
        audioStream.startWithURL("http://air.radiorecord.ru:8101/rr_128")
    }

    @IBAction func stopAction(sender: AnyObject) {
        print("stop")
        audioStream.stop()
    }
    
    //MARK: AudioStreamEngine delegate
    
    func audioStreamEngineDidUpdateTitle(title: String!) {
        
        print(title)
        
        titleLabel.stringValue = title
    }
}

